cmd/note_common.go

```diff
@@ -164,44 +170,65 @@
 			} else {
 				oldfile = f[4]
 			}
-		} else if strings.HasPrefix(scanner.Text(), "|") {
-			if comments != "" {
-				createCommitNote(project, mrID, commit, newfile, oldfile, oldLineNum, newLineNum, comments, block)
-				comments = ""
+			continue
+		}
+
+		// read line numbers
+		f := strings.Fields(strings.TrimSpace(scanner.Text()))
+		if f[1] == "@@" {
+			continue
+		}
+
+		oldLineNum = -1
+		newLineNum = -1
+
+		// The output can be, for example:
+		// | # # [no +/-] < context comment
+		// | # #    +     < newline comment
+		// | # #    -     < oldline comment
+		// | #      -     < oldline comment
+		// |   #    +     < newline comment
+
+		// f[0] is always a "|"
+		// f[1] will always be a number
+		val1, _ := strconv.Atoi(f[1])
+		val2, err := strconv.Atoi(f[2])
+		if err == nil { // f[2] is a number
+			if len(f) <= 3 { // f[3] does not exist
+				oldLineNum = val1
+				newLineNum = val2
+				linetype = "context"
+				continue
 			}
-			// read line numbers
-			fs := strings.Split(scanner.Text(), " ")
-
-			oldLineNum = -1
-			newLineNum = -1
-			for _, f := range fs {
-				if f == "" || f == "|" || f == "@@" {
-					continue
-				}
-				val, err := strconv.Atoi(f)
-				if err != nil {
-					// NaN
-					if strings.HasPrefix(f, "+") {
-						newLineNum = oldLineNum
-						oldLineNum = -1
-					}
-					break
-				} else {
-					// Number
-					if oldLineNum == -1 {
-						oldLineNum = val
-					} else {
-						newLineNum = val
-						break
-					}
-				}
+			// f[3] exists
+			newLineNum = val2
+			// f[3] starts with a +
+			if strings.HasPrefix(f[3], "+") {
+				linetype = "new"
+				continue
+			} else if strings.HasPrefix(f[3], "-") { // f[3] starts with a -
+				linetype = "old"
+				continue
```

Comment:
I think we need to set `oldLineNum` in this case.
Also in the "context" case (in addition to newLineNum). I believe we can just add a `oldLineNum = val1` next to `newLineNum = val1`

LGTM otherwise from a quick look.

Reply:
So something like this?  I think this should work.

```
diff --git a/cmd/note_common.go b/cmd/note_common.go
index be02a45afb21..17fd5386c86b 100644
--- a/cmd/note_common.go
+++ b/cmd/note_common.go
@@ -201,6 +201,7 @@ func createCommitComments(project string, mrID int, commit string, body string,
                                continue
                        }
                        // f[3] exists
+                       oldLineNum = val1
                        newLineNum = val2
                        // f[3] starts with a +
                        if strings.HasPrefix(f[3], "+") {

```

Reply:
Yes but seeing that we do the same in the if-statement above, we can just move these two lines before the if statement.

Reply:
Hmm, it doesn't appear to work that way.  If a new line comment is made the old line must be not set AFAICT.

Reply:
> Hmm, it doesn't appear to work that way. If a new line comment is made the old line must be not set AFAICT.

you're right, my bad